### PR TITLE
JCR Session handling optimization. Fixes #2320

### LINKED ISSUE #2320: Inefficient JcrSession handling
For a single requests (e.g. getModelConent) there are many jcr sessions opened and closed which causes high load on the DB. Session should be reused at least in http request scope.

In addition to that for each session which is opened there are additional requests to the DB to fetch the roles for the corresponding user.

### PR DESCRIPTION
**Pull request by t-gauss**
Signed-off-by: Tobias Gauss <tobias.gauss@bosch.io>

### CODE REVIEW

File: repository/repository-core/src/test/java/org/eclipse/vorto/repository/core/RequestRepositorySessionHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        // new session should be returned
+        Session sessionForReuse = helper.getSession();
+        Assert.assertFalse(mySession.equals(sessionForReuse));
+        // after destroy session should still be alive
+        helper.destroy();
+        Assert.assertTrue(sessionForReuse.isLive());
+
+    }
+
+    private Repository createMockRepository() {
```
Comment (kolotu) [inline repository/repository-core/src/test/java/org/eclipse/vorto/repository/core/RequestRepositorySessionHelperTest.java]: Wouldn't it be easier to use Mockito here? Or is there a reason not to use it for this test?

Comment (t-gauss) [reply repository/repository-core/src/test/java/org/eclipse/vorto/repository/core/RequestRepositorySessionHelperTest.java]: Hi @kolotu ,
thanks. I will adapt this.

Comment (mena-bosch) [reply repository/repository-core/src/test/java/org/eclipse/vorto/repository/core/RequestRepositorySessionHelperTest.java]: Can't check out the branch right now but I agree with the mock-related comment. Also (and forgive me if that's obvious) - are the invocation to `get` on the supplier always guaranteed to not return `null`?

Comment (t-gauss) [reply repository/repository-core/src/test/java/org/eclipse/vorto/repository/core/RequestRepositorySessionHelperTest.java]: Hi @mena-bosch ,
I didn't change the behaviour compared to the way it was before.

**Comment by t-gauss** [comment]
@kolotu Thanks. That looks much better readable.

**Pull request merged**